### PR TITLE
Where For Dinner TAP 1.4 Items

### PR DESCRIPTION
**Pull request by gm2552**
Covers AppSSO compliance issues for TAP 1.4 as well as enhancements.  Addresses issue #101

### CODE REVIEW

File: where-for-dinner/README.md [hunk 1]
```diff
@@ -52,3 +52,13 @@ Note that the event channel protocols are not specified as they can be swapped o
 The default build and deployment architecture uses Spring Cloud Streams as the asynchronous messaging implementation with RabbitMQ as the default binding.  The messaging topology is depicted in the following abridged architecture diagram: 
 
 ![](doc/images/SCSMessaging.png)
+
+### KNative Eventing
```
Comment (trisberg) [inline where-for-dinner/README.md:56]: Change to "Knative Eventing"

File: where-for-dinner/README.md [hunk 2]
```diff
@@ -52,3 +52,13 @@ Note that the event channel protocols are not specified as they can be swapped o
...
 
 ![](doc/images/SCSMessaging.png)
+
+### KNative Eventing 
+
+The *Cloud Events* option of the Where For Dinner application contains build configuration that removes Spring Cloud Streams dependencies and relies on the Spring Cloud Function webflux library for receiving events over HTTP.  Knative eventing fully embraces this paradigm and integrates seamlessly with functions exposed via the Spring Cloud Functions module.
+
+The architecture still uses Spring Cloud Streams as the messaging implementation for initially emitting searches, and a KNative `source` is inserted to create a bridge between the Spring Cloud Streams binding and the downstream Knative eventing concepts.  For the RabbitMQ implementation, a KNative RabbitMQ Source is declared to read events from a RabbitMQ exchange and propagate them through Knative eventing down to the subscribing services.  The following is a diagram of the Knative event concepts used for Where For Dinner:
+
+![](doc/images/KNativeEventing.png)
```
Comment (trisberg) [inline where-for-dinner/README.md:62]: Rename `KNativeEventing.png` to `KnativeEventing.png`

File: where-for-dinner/README.md [hunk 3]
```diff
@@ -52,3 +52,13 @@ Note that the event channel protocols are not specified as they can be swapped o
 The default build and deployment architecture uses Spring Cloud Streams as the asynchronous messaging implementation with RabbitMQ as the default binding.  The messaging topology is depicted in the following abridged architecture diagram: 
 
 ![](doc/images/SCSMessaging.png)
+
+### KNative Eventing 
+
+The *Cloud Events* option of the Where For Dinner application contains build configuration that removes Spring Cloud Streams dependencies and relies on the Spring Cloud Function webflux library for receiving events over HTTP.  Knative eventing fully embraces this paradigm and integrates seamlessly with functions exposed via the Spring Cloud Functions module.
+
+The architecture still uses Spring Cloud Streams as the messaging implementation for initially emitting searches, and a KNative `source` is inserted to create a bridge between the Spring Cloud Streams binding and the downstream Knative eventing concepts.  For the RabbitMQ implementation, a KNative RabbitMQ Source is declared to read events from a RabbitMQ exchange and propagate them through Knative eventing down to the subscribing services.  The following is a diagram of the Knative event concepts used for Where For Dinner:
```
Comment (trisberg) [inline where-for-dinner/README.md:60]: Change "KNative" to "Knative"

File: where-for-dinner/accelerator.yaml [hunk 4]
```diff
@@ -70,25 +61,46 @@ accelerator:
...
-      name: dbType
-      value: mysql
-  - label: Create Resource Claim Defintion for Services
-    description: "If set to true, a resource claim definition yaml file for the data services will be created."
-    defaultValue: true
-    name: createResourceClaim
+      name: cacheType
+      value: redisCache
+  - label: Enable Cloud Events
+    description: "If set to true, the application will use CloudEvents and KNative as the internal eventing/messagiung model instead of spring cloud streams."
```
Comment (trisberg) [inline where-for-dinner/accelerator.yaml:86]: KNative should be Knative

File: where-for-dinner/accelerator.yaml [hunk 5]
```diff
@@ -70,25 +61,46 @@ accelerator:
...
+    defaultValue: false
+    name: enableCloudEvents
     inputType: checkbox
     dataType: boolean
     required: true
-  - label: Enable Security (Experimental)
-    description: "If set to true, security will enabled on the application and APIs.  You will need to provide additional security configuation in the subsequent sections below."
+  - label: Use RabbitMQ KNative Eventing Broker
```
Comment (trisberg) [inline where-for-dinner/accelerator.yaml:92]: KNative should be Knative

File: where-for-dinner/doc/TAPDeployment.md [hunk 6]
```diff
@@ -185,19 +198,17 @@ The accelerator contains the following configuration options:
...
-* **Redirect URI:**  If security is enabled, this is the redirect URI back the API gateway service after a successful user authentication.
+* **Database Name:**  The name of the database resource that the micro-services will connect to.  This name will be propagated to the `workload.yaml` files in the resource claim section to indicate the names of the database that micro-services should connect to.
+* **Enable Cloud Events:** If this box is checked, the accelerator will generate a configuration yaml file in the `config/app-operator` directory that contains the resource definitions for the KNative eventing resources.  These include the RabbitMQ source as well as the broker and trigger configurations for routing events to downstream services.  This will also configure applicable services to remove the Spring Cloud Stream binding libraries from the build process.
```
Comment (trisberg) [inline where-for-dinner/doc/TAPDeployment.md:204]: KNative should be Knative

File: where-for-dinner/doc/TAPDeployment.md [hunk 7]
```diff
@@ -185,19 +198,17 @@ The accelerator contains the following configuration options:
...
+* **Use RabbitMQ KNative Eventing Broker:** If this box is checked, the KNative broker will use a RabbitMQ broker implementation.
```
Comment (trisberg) [inline where-for-dinner/doc/TAPDeployment.md:205]: KNative should be Knative

**Review by trisberg — CHANGES_REQUESTED** [review]
I also see a bunch of files with `hungryman` in the name, we might want to rename them

```
├── catalog
│   ├── components
│   │   ├── hungryman-api-gateway
│   │   │   └── hungryman-api-gateway.yaml
│   │   ├── hungryman-availability
│   │   │   └── hungryman-availability.yaml
│   │   ├── hungryman-notify
│   │   │   └── hungryman-notify.yaml
│   │   ├── hungryman-search
│   │   │   └── hungryman-search.yaml
│   │   ├── hungryman-search-proc
│   │   │   └── hungryman-search-proc.yaml
│   │   └── hungryman-ui
│   │       └── hungryman-ui.yaml
│   ├── resources
│   │   ├── db
│   │   │   └── hungryman-db-resource.yaml
│   │   └── messaging
│   │       └── hungryman-messaging-resource.yaml
│   └── systems
│       └── hungryman-system.yaml

├── doc
│   ├── TAPDeployment.md
│   └── images
│       ├── AppHomeScreen.png
│       ├── HungrymanHighLevelArch.png
│       ├── KNativeEventing.png
│       └── SCSMessaging.png

├── icons
│   └── hungryman.png
```

File: where-for-dinner/doc/TAPDeployment.md (hunk 7 above)
Comment (gm2552) [reply where-for-dinner/doc/TAPDeployment.md:205]: The catalog items may need to fixed for 1.3.1 as w. ll. The catalog updates that were done for 1.3.0 renamed the references to the directories to "where-for-dinner" but did not actually change the directory names.

File: where-for-dinner/README.md (hunk 1 above)
Comment (gm2552) [reply where-for-dinner/README.md]: Updated.

File: where-for-dinner/README.md (hunk 2 above)
Comment (gm2552) [reply where-for-dinner/README.md]: Updated.  Reference to updated as well in documentation.

File: where-for-dinner/README.md (hunk 3 above)
Comment (gm2552) [reply where-for-dinner/README.md]: Updated.

File: where-for-dinner/accelerator.yaml (hunk 5 above)
Comment (gm2552) [reply where-for-dinner/accelerator.yaml]: Updated.

File: where-for-dinner/doc/TAPDeployment.md (hunk 6 above)
Comment (gm2552) [reply where-for-dinner/doc/TAPDeployment.md]: Updated.

File: where-for-dinner/doc/TAPDeployment.md (hunk 7 above)
Comment (gm2552) [reply where-for-dinner/doc/TAPDeployment.md]: Catalog file and directory names have been updated in this PR.

**Review by trisberg — APPROVED** [review]

**Pull request merged**